### PR TITLE
[WIP] Update the robot state helper for ROS2

### PR DESCRIPTION
**Pull request by livanov93**
Implementing convinient class for easy communication with robot for ROS2.

### CODE REVIEW

File: ur_robot_driver/src/robot_state_helper.cpp [hunk 1]
```diff
@@ -27,7 +27,316 @@
...
+                                                                         << robotModeString(urcl::RobotMode::IDLE)
+                                                                         << " once the teach button is released.");
+            break;
+          case urcl::RobotMode::RUNNING:
+            RCLCPP_INFO_STREAM(logger, "The robot has reached operational mode " << robotModeString(robot_mode_));
+            break;
+          default:
+            RCLCPP_WARN_STREAM(logger, "The robot is currently in mode " << robotModeString(robot_mode_)
+                                                                         << ". This won't be handled by this helper. "
+                                                                            "Please "
```
Comment (destogl) [inline ur_robot_driver/src/robot_state_helper.cpp:225]: This is a strange layout....

File: ur_robot_driver/src/robot_state_helper.cpp [hunk 2]
```diff
@@ -27,7 +27,316 @@
...
+  {
+    RCLCPP_ERROR(rclcpp::get_logger("ur_robot_state_helper"), "Failed to call dashboard trigger service ");
+  }
+
+  if (!result.get()->success)
+  {
+    result_.success = false;
+    result_.message = "An error occurred while switching the robot mode. The respective dashboard service returned "
+                      "non-success: " +
+                      result.get()->message;
```
Comment (destogl) [inline ur_robot_driver/src/robot_state_helper.cpp:254]:
```suggestion
                      "non-success: " + result.get()->message;
```

**Review by destogl — COMMENTED** [review]
This looks fine, but I am not sure how it should work in ros2_control.... @livanov93 can you comment?

**Comment by destogl** [comment]
@livanov93 Is this sill used?

**Comment by livanov93** [comment]
In general, in the ROS1 version, this was a convenient node for interacting with the robot via services and simple SimpleActionClient/Server that are registered in the "main" driver node. However, the biggest problem was that there was no SimpleActionClient/Server abstraction in the ROS2 at the start of this PR. This was the reason the PR development was postponed. It can be done with the existing ROS2 actionlib but the logic within the node will get more complex than it was before.

**Comment by fmauch** [comment]
Closing this in favor of #933

**Pull request closed without merging**